lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -506,11 +483,13 @@ computeTxSize proxy pm witTag action cs =
     bloatChaff :: Word -> BL.ByteString -> ByteString
     bloatChaff n = BL.toStrict . BL.take (fromIntegral n) . BL.cycle
 
+    -- TODO: Surely we can allow byron witnesses paying for certificates?
+    -- Should be no reason to case here.
```

Comment:
Actually we do not. We don't have hybrid wallets, so a wallet that would require Byron witnesses is by construction a "Byron wallet" with no delegation / rewards capability.

Reply:
To me it seems like there is not benefit to not allowing it, and limits our options for e.g. integration test setup.

Reply:
The benefit is lowering complexity. There are a lot of things that hold together precisely because only Shelley wallets are able to delegate and have withdrawals. Changing this would add a great deal of complexity AND, it'll remove the incentive there's for people to move to Shelley wallets (remember that Byron wallets have security flaws).

Reply:
Changing `mkTx` should have no user-facing effect since shelley wallets' won't select byron inputs.

I also suspect the code here would be more elegant, and we would have the option to use a slightly more flexible `mkTx` in integration tests / for scripts.

Reply:
From within the transaction layer it shouldn't really have any visible impact indeed. And it's perhaps best to not make any assumption about the "outside context" within the transaction layer actually.